packages/rolldown/src/cli/load-config.ts

```diff
@@ -76,14 +81,57 @@ async function findConfigFileNameInCwd(): Promise<string> {
 }
 
 export async function loadTsConfig(configFile: string): Promise<ConfigExport> {
-  const file = await bundleTsConfig(configFile)
+  const isEsm = isFilePathESM(configFile)
+  const file = await bundleTsConfig(configFile, isEsm)
   try {
     return (await import(pathToFileURL(file).href)).default
   } finally {
     fs.unlink(file, () => {}) // Ignore errors
   }
 }
 
+export function isFilePathESM(filePath: string): boolean {
+  if (/\.m[jt]s$/.test(filePath)) {
+    return true
+  } else if (/\.c[jt]s$/.test(filePath)) {
+    return false
+  } else {
+    // check package.json for type: "module"
+    try {
+      const pkg = findNearestPackageData(path.dirname(filePath))
```

Comment:
There is no need to try again, since you already ignore error in `findNearestPackageData`

Reply:
Fixed it.